lang_GENERIC_base/AST_generic.ml

```diff
@@ -505,6 +505,7 @@ and special =
   (* special vars *)
   | This | Super (* called 'base' in C# *)
   | Self | Parent (* different from This/Super? *)
+  | NextArrayIndex (* Lua *)
```

Comment:
I'd rather find whether we could reuse another constructor to represent that.
There's probably an equivalent in other programming languages.
What is it used for?